### ISSUE TITLE
[FEA] Sort list columns

### ISSUE THREAD

**Issue by shwina** [rapidsai/cudf#5890]
It would be nice to be able to sort list columns. The use case I have encountered so far is in testing equality of two list columns with identical elements, but ordered differently, as occurs in a [`groupby COLLECT`](https://github.com/rapidsai/cudf/pull/5874). 

## What does it mean to sort a list column?

Python does a "lexical" comparison between lists:

```python
>>> compare([1, 2], [1, 2])
[1, 2] == [1, 2]

>>> compare([1, 2], [1, 2, 3])
[1, 2] < [1, 2, 3]

>>> compare([1, 2], [1, 1, 3])
[1, 2] > [1, 1, 3]
```

Accepting the above, it becomes easy to reason about comparing arbitrarily nested lists:

```python
>>> compare([[[1, 2], [1, 3]], [[4, 5]]], [[[1, 2], [1, 3]], [[4, 6]]])
[[[1, 2], [1, 3]], [[4, 5]]] < [[[1, 2], [1, 3]], [[4, 6]]]

>>> compare([[[1, 2], [1, 4]], [[4, 5]]], [[[1, 2], [1, 3]], [[4, 6]]])
[[[1, 2], [1, 4]], [[4, 5]]] > [[[1, 2], [1, 3]], [[4, 6]]]
```

I wonder if Spark users would have similar expectations for sorting list columns. cc: @nvdbaranec @jlowe @jrhemstad @kkraus14 @harrism

**Comment by kkraus14** [comment]
cc @revans2 from the Spark side as well
cc @felipeblazing @williamBlazing for if you have thoughts / opinions from the BlazingSQL side

**Comment by jrhemstad** [comment]
So even if we decide on what it means to sort a list of columns, implementing this in the context of a multi-column sort is going to be very difficult. What's more, I'm concerned adding the logic to the `row_comparator` could cause a performance regression in multi-column sorting even in the absence of list columns.

**Comment by jlowe** [comment]
Spark has a similar comparison for list types.  There may be some discrepancies when null entries within a list are encountered.  Spark treats a null entry within a list as less than any other value in the other list except another null which is treated as equal.

> I'm concerned adding the logic to the row_comparator could cause a performance regression in multi-column sorting even in the absence of list columns.

Can you elaborate?  I'm curious if this will be like NaN handling, where the performance hit was negligible unless NaNs appeared in the data.

**Comment by jrhemstad** [comment]
> Can you elaborate? I'm curious if this will be like NaN handling, where the performance hit was negligible unless NaNs appeared in the data.

It will be similar to the AST interpretation Brad is doing where adding increasingly complex and nested switch statements eventually causes the compiler to balk and starts putting things on the stack frame or refusing to inline things, which significantly reduces performance. Even if list columns aren't present, the increased complexity of the comparator could very well reduce performance.

**Comment by nvdbaranec** [comment]
Still relevant.  And I believe we will be starting to investigate in the somewhat-near future.

**Comment by GregoryKimball** [comment]
Closed by #11129
